rust/src/connection.rs

```diff
@@ -7,7 +7,11 @@ use anyhow::Result;
 pub async fn keep_alive() -> Result<()> {
     let mut connected = false;
     loop {
-        let is_connection_lost_maybe = !is_first_channel_usable()?;
+        let channel_details = wallet::get_first_channel_details();
+
+        let is_connection_lost_maybe = channel_details.is_some()
+            && !is_first_channel_usable()
+            && channel_details.unwrap().short_channel_id.is_some();
```

Comment:
I would not unwrap here but use `unwrap_or_default` which evaluates to false.

Reply:
Second thought: `            && channel_details.unwrap().short_channel_id.is_some();`
should be redundant as `is_first_channel_usable`, i.e. `channel.is_usable` is only true if the channel is confirmed.

Reply:
Third thought: I think this does not work as you want.

if we do have a channel which is currently being confirmed `is_first_channel_usable=false` and `channel_details.unwrap().short_channel_id.is_some()=false`  then we would constantly reconnect.

Reply:
Maybe I'm missing something, but do we even need this logic. 
In `connect_peer_if_necessary` we check if the peer is known and hence we have a connection: 

https://github.com/itchysats/10101/blob/576e51a8cff6668ac1cf57f8f32584d0d8c0b124/rust/src/lightning.rs#L637-L641

Reply:
cc @da-kami / @klochowicz

Reply:
My understanding:

We used `connect_peer_if_necessary` and it did not work stable, i.e. we encountered problems with the connection when e.g. opening a channel before. I don't know if we can rely on the `peer_manager` to decide if the connection is healthy - problems that we encountered in the past would point to no.

That's why additional logic is being added to decide if the connection is still considered healthy.

Reply:
basically, I would change this logic to [`477e8e2` (#496)](https://github.com/itchysats/10101/pull/496/commits/477e8e2bb07f9b386492e2f69bc23916ddcd3fd1)